staging/src/k8s.io/component-base/logs/logs.go

```diff
@@ -168,6 +168,16 @@ func (writer KlogWriter) Write(data []byte) (n int, err error) {
 // InitLogs initializes logs the way we want for Kubernetes.
 // It should be called after parsing flags. If called before that,
 // it will use the default log settings.
+//
+// InitLogs disables support for contextual logging in klog while
+// that Kubernetes feature is not considered stable yet. Commands
+// which want to support contextual logging can:
+// - call klog.EnableContextualLogging after calling InitLogs,
+//   with a fixed `true` or depending on some command line flag or
+//   a feature gate check
+// - set up a FeatureGate instance, the advanced logging configuration
+//   with Options and call Options.ValidateAndApply with the FeatureGate;
+//   k8s.io/component-base/logs/example/cmd demonstrates how to do that
 func InitLogs() {
```

Comment:
does this play nicely with the way component-base CLI repeatedly calls InitLogs()?

https://github.com/kubernetes/kubernetes/blob/f25c0e5f094485f710e80377fbb833d3eeed7492/staging/src/k8s.io/component-base/cli/run.go#L122-L144

Reply:
It should only get injected into one of the steps, i.e. only get called once. The sequence is `main -> parse flags [+ config] -> InitLogs [-> ValidateAndApply]`.

If someone calls `InitLogs` again later, it will disable contextual logging again - probably not intended, but failing "safely" in the sense that the feature is off.